test/test_grade/test_integration.py

```diff
@@ -362,3 +364,24 @@ def test_config_overrides_integration():
     got = got.reindex(sorted(got.columns), axis=1)
     want = want.reindex(sorted(want.columns), axis=1)
     assert got.equals(want)
+
+
+@pytest.mark.slow
+@pytest.mark.docker
+def test_grade_summaries():
+    """
+    Checks that are grading summaries are written to the disck
+    """
+    grade(
+        name = ASSIGNMENT_NAME,
+        paths = [FILE_MANAGER.get_path("notebooks")],
+        output_dir = "test/",
+        autograder = AG_ZIP_PATH,
+        summaries = True
+    )
+    for filename in os.listdir(FILE_MANAGER.get_path("notebooks")):
+        file_path = os.path.join("test/grading-summaries", filename)
+        if os.path.isfile(file_path):
+            assert os.path.exists(file_path)
+            with open(file_path, 'r') as file:
+                assert file.read()
```

Comment:
Regardless of what's being written, this test should check that the contents are correct.

Reply:
PTAL at this as well.

Reply:
Good deal -- checking the contents of summary files now

Reply:
I am not quite sure how you would mock this -- the summary files are written in grade/__init__.py main function. I am checking the files that are written to the disk. Maybe I am missing something here?

Reply:
Since the changes are in `__init__.py`, you can mock out the `launch_containers` function so it doesn't actually run but instead just returns some predetermined results. There's an example of this in `test_config_overrides` above (the mock is created with the `@mock.patch(...)` decorator and the return value is set by setting `mocked_launch_grade.return_value`.

Reply:
Ok, wait I am with you. Give me a second

Reply:
I rebased and forced the last commit to prevent another commit here.

I think I got it. Good deal